cran/utils.go

```diff
@@ -87,3 +129,37 @@ func RepoURLHash(r RepoURL) string {
 	urlHash := fmt.Sprintf("%x", h.Sum(nil))
 	return r.Name + "-" + urlHash[:12]
 }
+
+func ReadOsRelease() {
+	if osRelease != nil {
+		// Already cached
+		return
+	}
+	//os-release doesn't consistently quote variables, so we need to manipulate it a little bit
+	configData, err := ioutil.ReadFile("/etc/os-release")
+
+	//Find all unquoted strings and quote them
+	re := regexp.MustCompile(`(.*?=)([^"].*)`)
+	fixedConfig := re.ReplaceAll(configData, []byte("${1}\"${2}\""))
+
+	//Let viper map it in otherwise it defaults to yaml
+	//Don't pollute the global viper instance or we'll segfault later trying to read pkgr.yml
+	vp := viper.New()
```

Comment:
I'm uncertain why the use of viper vs just using the toml unmarshaller directly? 

```
	re := regexp.MustCompile(`(.*?=)([^"].*)`)
	osRel, err := ioutil.ReadFile("os-release")
	if err != nil {
		panic(err)
	}
	fixedConfig := re.ReplaceAll(osRel, []byte("${1}\"${2}\""))
	osr := OsRelease{}
	err = toml.Unmarshal(fixedConfig, &osr)
	if err != nil {
		log.Fatal("could not unmarshall /etc/os-release with err: %v", err)
	}
```

Reply:
TOML unmarshaling is less flexible than Viper (which modifies it somewhat), and /etc/os-release fields are not consistent across distros, which is why there are some pointers (for nullable fields)